### PR TITLE
Add explicit scope to the new repo form

### LINKED ISSUE #2995: No obvious way to create global app repository
### Description:

Currently if you click on the option to add an app repository, you are presented with:

create-repos

showing both global app repositories and those for the current namespace (if any). It is completely non-obvious how you create a global repository. The message displayed there used to encourage the user to select "All Namespaces" for the old namespace selector, but the concept of "All namespaces" was removed when we moved to the context selector, so now you need to know to select the "kubeapps" (or namespace in which Kubeapps is installed) to then add a global namespace.

Originally reported by Scott from customer reports.

### PR DESCRIPTION
**Pull request by antgamdia**
### Description of the change

Adding an explicit scope for selecting whether the repo should be namespace or global + some info messages to the user.

### Benefits

Adding a global repo will no longer require switching the context, which wasn't something trivial to figure out as a user.

### Possible drawbacks

N/A

### Applicable issues

- fixes #2995 

### Additional information

reposWithScope

Squeezing two changes:
- Disable the description field if using carvel repos (as it currently errors if any is passed)
- Fix some <label> tags to comply with a11y recommendations.

### CODE REVIEW

**Comment by ppbaena** [comment]
Nice to see this UX improvement when configuring repositories. Thanks @antgamdia!

File: dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx [hunk 1]
```diff
@@ -637,6 +676,7 @@ export function PkgRepoForm(props: IPkgRepoFormProps) {
                     placeholder="Description of the repository"
                     value={description || ""}
                     onChange={handleDescriptionChange}
+                    disabled={plugin?.name === (PluginNames.PACKAGES_KAPP as string)}
```
Comment (antgamdia) [inline dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx:679]: Disabling description if the plugin is kapp

File: dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx [hunk 2]
```diff
@@ -686,143 +732,210 @@ export function PkgRepoForm(props: IPkgRepoFormProps) {
...
+                      {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+                      <label>Scope:</label>
+                      <CdsControlMessage>
+                        If the repository is global, its packages will be availabe in every
+                        namespace of the cluster. If it is namespaced-scoped, its packages will be
+                        available only in the namespace where the repository has been installed.
+                      </CdsControlMessage>
+
+                      {repo.namespaceScoped ||
+                        (isGlobalNamespace(namespace, plugin?.name, currentNsConfig) && (
```
Comment (antgamdia) [inline dashboard/src/components/Config/PkgRepoList/PkgRepoForm.tsx:746]: Using either the field from the repo object (editing the existing repo) or calculating if the ns is global or not.

File: dashboard/src/shared/utils.ts [hunk 3]
```diff
@@ -235,3 +235,21 @@ export function isGlobalNamespace(namespace: string, pluginName: string, kubeapp
       return false;
   }
 }
+
+export function getGlobalNamespaceOrNamespace(
```
Comment (antgamdia) [inline dashboard/src/shared/utils.ts:239]: This should belong to each plugin's config

**Review by castelblanque — APPROVED** [review]
LGTM

**Pull request merged**